esp/services/ws_decoupledlogging/ws_decoupledloggingservice.hpp

```diff
@@ -33,13 +33,18 @@ class CWSDecoupledLogSoapBindingEx : public CWSDecoupledLogSoapBinding
         : CWSDecoupledLogSoapBinding(cfg, bindname, procname, level) { };
 };
 
+struct WSDecoupledLogAgentGroup
+{
+    StringAttr name;
+    StringAttr tankFileDir, tankFileMask;
+    std::vector<IUpdateLogThread*>  loggingAgentThreads;
+};
+
 class CWSDecoupledLogEx : public CWSDecoupledLog
 {
-    StringAttr espProcess, tankFileDir;
+    StringAttr espProcess;
     IEspContainer* container;
-
-    typedef std::vector<IUpdateLogThread*> LOGGING_AGENTTHREADS;
-    LOGGING_AGENTTHREADS  loggingAgentThreads;
+    std::vector<WSDecoupledLogAgentGroup*> logGroups;
```

Comment:
What benefit is obtained by making this a collection of pointers instead of structures? As is, a destructor is needed to delete the contained instances.

Yes, the lifetime of the vector is the lifetime of the service, which is expected to be the lifetime of the process. It's just that a very small change can avoid adding to the noise generated by tools like valgrind when they report "leaks" that really aren't.